container/container.go

```diff
@@ -324,6 +324,11 @@ func (container *Container) CheckpointDir() string {
 // StartLogger starts a new logger driver for the container.
 func (container *Container) StartLogger() (logger.Logger, error) {
 	cfg := container.HostConfig.LogConfig
+
+	if cfg.Type == "none" {
+		return nil, logger.ErrReadLogsNotSupported
```

Comment:
Moved it to [daemon/logs.go#L33](https://github.com/docker/docker/blob/58028a2919f1ee794a058fb3451a5145a6382b4d/daemon/logs.go#L33) as discussed on slack